scheduler/kubernetes/snaps-3d.md

```diff
@@ -44,62 +58,127 @@ spec:
+Below post-snapshot rule reduces the lock taken by db.fsyncLock() on a mongod instance by 1. (See [reference](https://docs.mongodb.com/manual/reference/method/db.fsyncUnlock/#db.fsyncUnlock))
 ```
 apiVersion: stork.libopenstorage.org/v1alpha1
 kind: Rule
 metadata:
-  name: px-hello-world-rule
+  name: px-mongodb-postsnap-rule
 spec:
   - podSelector:
-      app: foo
+      app: px-mongo-mongodb
     actions:
     - type: command
-      value: echo "hello world"
-      runInSinglePod: true
+      value: mongo --eval "printjson(db.fsyncUnlock())"
 ```
 
-#### 2. Create VolumeSnapshots that reference the rules
+**Snapshot**
 
-Once you have the rules applied in your cluster, you can reference them in the `VolumeSnapshot` using the following annotations.
+Creating the below VolumeSnapshot will do the following:
 
-* __stork.rule/pre-snapshot__: Stork will execute the rule which is given in the value of this annotation _before_ taking the snapshot.
-* __stork.rule/post-snapshot__: Stork will execute the rule which is given in the value of this annotation _after_ taking the snapshot.
+* Stork will run the _px-mongodb-presnap-rule_ rule on pod using the _px-mongo-pvc_ PVC.
+* Once the rule is executed, Stork will take a snapshot of the _px-mongo-pvc_ PVC.
+* After the snapshot has been triggered, Stork will run the _px-mongodb-postsnap-rule_ rule on pod using the _px-mongo-pvc_ PVC.
+
+```
+apiVersion: volumesnapshot.external-storage.k8s.io/v1
+kind: VolumeSnapshot
+metadata:
+  name: mongodb-3d-snapshot
+  annotations:
+    stork.rule/pre-snapshot: px-mongodb-presnap-rule
+    stork.rule/post-snapshot: px-mongodb-postsnap-rule
+spec:
+  persistentVolumeClaimName: px-mongo-pvc
+```
 
-**Examples**
+### Cassandra
 
-Follow is an example of a VolumeSnapshot which will do the following:
+**Pre-snapshot rule**
 
-* Stork will run the _px-presnap-rule_ rule on all pods that are using PVCs that match labels _app=mysql_.
-* Once the rule is executed, Stork will take a snapshot of all PVCs that match labels _app=mysql_. Hence this will be a group snapshot.
-* After the snapshot has been triggered, Stork will terminate any background actions that may exist in the rule _px-presnap-rule_.
+Below rule flushes the tables from the memtable on all cassandra pods.
+```
+apiVersion: stork.libopenstorage.org/v1alpha1
+kind: Rule
+metadata:
+  name: px-cassandra-rule
+spec:
+  - podSelector:
+      app: cassandra
+    actions:
+    - type: command
+      value: nodetool flush
+```
+
+**Snapshot**
 
 ```
 apiVersion: volumesnapshot.external-storage.k8s.io/v1
 kind: VolumeSnapshot
 metadata:
-  name: mysql-3d-snapshot
-  namespace: default
+  name: cassandra-3d-snapshot
   annotations:
-    portworx.selector/app: mysql
-    stork.rule/pre-snapshot: px-presnap-rule
+    portworx.selector/app: cassandra
+    stork.rule/pre-snapshot: px-cassandra-rule
 spec:
-  persistentVolumeClaimName: mysql-data-1
+  persistentVolumeClaimName: cassandra-data-1
+```
+
+### Hello world
```

Comment:
This as the first generic example would be better than at the end.

Reply:
done